File: src/0960-Delete-Columns-to-Make-Sorted-III/0960.cpp
static int x = []() {std::ios::sync_with_stdio(false); cin.tie(0); return 0; }();
class Solution 
{
public:
    int minDeletionSize(vector<string>& A) 
    {
        vector<int> mem(A[0].size(), 1);
        for (int i = 0; i < A[0].size(); ++i) 
        {
            for (int j = 0; j < i; ++j)
            {
                for (int k = 0; k <= A.size(); ++k) 
                {
                    if (k == A.size()) mem[i] = max(mem[i], mem[j] + 1);
                    else if (A[k][j] > A[k][i]) break;
                }               
            }
        }
        return A[0].size() - *max_element(begin(mem), end(mem));
    }
};

File: src/0960-Delete-Columns-to-Make-Sorted-III/0960.py
class Solution:
    def minDeletionSize(self, A):
        """
        :type A: List[str]
        :rtype: int
        """
        r, c = len(A), len(A[0])
        mem = [1] * c
        for i in range(1, c):
            for j in range(i):
                for k in range(r+1):  
                    if k == r:
                        mem[i] = max(mem[i], mem[j] + 1)
                    elif A[k][j] > A[k][i]:
                        break
                        
        return c - max(mem)